src/main/java/org/elasticsearch/cluster/action/index/NodeIndexDeletedAction.java

```diff
@@ -65,29 +65,18 @@ public void remove(Listener listener) {
 
     public void nodeIndexDeleted(final ClusterState clusterState, final String index, final String nodeId) throws ElasticsearchException {
         DiscoveryNodes nodes = clusterState.nodes();
+        // TODO consolidate these all into one call but for now this is much easier to backport and we can do it in a second PR
         if (nodes.localNodeMaster()) {
             threadPool.generic().execute(new Runnable() {
                 @Override
                 public void run() {
                     innerNodeIndexDeleted(index, nodeId);
+                    innerNodeIndexStoreDeleted(index, nodeId);
                 }
             });
         } else {
             transportService.sendRequest(clusterState.nodes().masterNode(),
                     INDEX_DELETED_ACTION_NAME, new NodeIndexDeletedMessage(index, nodeId), EmptyTransportResponseHandler.INSTANCE_SAME);
-        }
-    }
-
-    public void nodeIndexStoreDeleted(final ClusterState clusterState, final String index, final String nodeId) throws ElasticsearchException {
```

Comment:
do we want to unify this with nodeIndexDeleted? I think it's better to have this called once the store is deleted in IndicesService#deleteShardStore .